File: src/main/java/work/lclpnet/ap2/impl/util/scoreboard/ScoreboardLayout.java
package work.lclpnet.ap2.impl.util.scoreboard;

public class ScoreboardLayout {

    public static final int UNDEFINED = -1;
    public static final int TOP = 0;
    public static final int BOTTOM = 1;
    private int topId = Integer.MAX_VALUE - 1;
    private int bottomId = Integer.MIN_VALUE + 1;

    public void addTop(ScoreHandle handle) {
        handle.setScore(topId--);
    }

    public void addBottom(ScoreHandle handle) {
        handle.setScore(bottomId++);
    }
}

File: src/lib/src/main/kotlin/work/lclpnet/ap2/util/scoreboard/ScoreboardLayout.kt
package work.lclpnet.ap2.util.scoreboard

class ScoreboardLayout {
    private var topId = Int.MAX_VALUE - 1
    private var bottomId = Int.MIN_VALUE + 1

    fun addTop(): Int = topId--

    fun addBottom(): Int = bottomId++

    fun resolvePosition(position: Int): Int = when (position) {
        TOP -> addTop()
        BOTTOM -> addBottom()
        else -> 0
    }

    companion object {
        const val TOP: Int = 0
        const val BOTTOM: Int = 1
    }
}
